pkg/client/logs_test.go

```diff
@@ -297,24 +301,6 @@ func TestPodsForLogs(t *testing.T) {
 			expectedPodCount:       1,
 			expectedError:          errors.New("failed to list pods: error"),
 		},
-		{
-			desc:                   "Unknown plugin specified results in plugin pods being fetched, then retry with all pods fetched",
-			pluginName:             "unknown-plugin",
-			expectedCallCount:      2,
-			expectedLabelSelectors: []string{"sonobuoy-plugin=unknown-plugin", ""},
-			podListErrors:          []error{nil, nil},
-			expectedPodCount:       2,
-			expectedError:          nil,
-		},
-		{
-			desc:                   "Error when fetching plugin pods on retry results in error being returned",
-			pluginName:             "unknown-plugin",
-			expectedCallCount:      2,
-			expectedLabelSelectors: []string{"sonobuoy-plugin=unknown-plugin", ""},
-			podListErrors:          []error{nil, errors.New("error")},
-			expectedPodCount:       2,
-			expectedError:          errors.New("failed to list pods: error"),
-		},
 	}
 
 	for _, tc := range testCases {
```

Comment:
Huh, I can't seem to add a comment to folded lines in the diff so I will add it here :confused: 

These test cases can be simplified a lot now given that we only ever call `List` once. Anything related to `callCount` can be removed, which means that `expectedLabelSelectors` and `podListErrors` no longer need to be arrays, they can just be a string and error.